platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/Rect.ts

```diff
@@ -13,14 +13,35 @@ class Rect {
 		this.width = width;
 		this.height = height;
 	}
+	
+	public right() {
+		return this.x + this.width;
+	}
+
+	public bottom() {
+		return this.y + this.height;
+	}
 
 	public static fromDomRect(rect: DOMRect) {
 		return new Rect(rect.x, rect.y, rect.width, rect.height);
 	}
 	
 	public clamp(point: Vector2d) {
 		return new Vector2d(
-			clamp(point.x, this.x, this.x + this.width - 1),
-			clamp(point.y, this.y, this.y + this.height - 1));
+			clamp(point.x, this.x, this.right() - 1),
+			clamp(point.y, this.y, this.bottom() - 1));
+	}
+	
```

Comment:
The methods `getMergedWith` and `contains` are currently unused, but this class is more of a library class, so I would keep them, because they might be useful later.